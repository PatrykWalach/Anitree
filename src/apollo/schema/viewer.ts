export interface User {
  id: number
  name: string
  avata: UserAvatar
  siteUrl: string
  options: {
    profileColor: string
    displayAdultContent: boolean
  }
  mediaListOptions: {
    scoreFormat: ScoreFormat
    animeList: MediaListTypeOptions
    mangaList: MediaListTypeOptions
  }
}
export interface UserAvatar {
  large: string
  medium: string
}
export type ScoreFormat =
  | 'POINT_100'
  | 'POINT_10_DECIMAL'
  | 'POINT_10'
  | 'POINT_5'
  | 'POINT_3'

export interface MediaListTypeOptions {
  customLists: string[]
  advancedScoring: string[]
  advancedScoringEnabled: boolean
}
