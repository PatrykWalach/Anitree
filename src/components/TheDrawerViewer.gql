fragment TheDrawerViewer_viewer on User {
  id
  name
  avatar {
    large
  }
  siteUrl
}
