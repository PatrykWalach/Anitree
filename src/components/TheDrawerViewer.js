import gql from 'graphql-tag'

export const TheDrawerViewer_viewer = gql`
  fragment TheDrawerViewer_viewer on User {
    id
    name
    avatar {
      large
    }
    siteUrl
  }
`
