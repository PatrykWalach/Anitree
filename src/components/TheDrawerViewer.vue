<template>
  <v-list-group active-class="accent--text" color="red">
    <template v-slot:activator>
      <v-list-item-avatar min-width="32" width="32" height="40">
        <v-img :src="viewer.avatar.large">
          <template v-slot:placeholder>
            <v-skeleton-loader type="image" />
          </template>
        </v-img>
      </v-list-item-avatar>
      <v-list-item-content>
        <v-list-item-title>
          {{ viewer.name }}
        </v-list-item-title>
      </v-list-item-content>
    </template>

    <v-list-item rel="noopener" :href="`${viewer.siteUrl}/animelist`">
      <v-list-item-icon>
        <v-icon>movie</v-icon>
      </v-list-item-icon>
      <v-list-item-content>
        <v-list-item-title>
          Anime List
        </v-list-item-title>
      </v-list-item-content>
    </v-list-item>

    <v-list-item rel="noopener" :href="`${viewer.siteUrl}/mangalist`">
      <v-list-item-icon>
        <v-icon>menu_book</v-icon>
      </v-list-item-icon>
      <v-list-item-content>
        <v-list-item-title>
          Manga List
        </v-list-item-title>
      </v-list-item-content>
    </v-list-item>

    <v-list-item @click.stop="logout">
      <v-list-item-icon>
        <v-icon>exit_to_app</v-icon>
      </v-list-item-icon>
      <v-list-item-content>
        <v-list-item-title>
          Logout
        </v-list-item-title>
      </v-list-item-content>
    </v-list-item>
    <!-- <v-divider></v-divider> -->
  </v-list-group>
</template>
<script lang="ts">
import { createComponent } from '@vue/composition-api'
import { settingsActions } from '../store/reducers/settings'
import { useDispatch } from 'vue-redux-hooks'

export default createComponent({
  inheritAttrs: false,
  props: {
    viewer: { default: null, required: true, type: Object },
  },
  setup() {
    const dispatch = useDispatch()

    const logout = () => {
      dispatch(settingsActions.CHANGE_TOKEN(null))
    }

    return { logout }
  },
})
</script>
