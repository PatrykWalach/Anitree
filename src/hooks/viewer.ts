import { inject } from '@vue/composition-api'
import { useQuery } from '@vue/apollo-composable'
export const DefaultViewer = Symbol('Viewer')

export const useViewer = <V = ReturnType<typeof useQuery>>() =>
  inject(DefaultViewer) as V
