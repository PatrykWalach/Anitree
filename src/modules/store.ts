import * as modules from '.'
import { Store, plugin } from 'vuex-composition-api'

export const store =
  process.env.NODE_ENV === 'production'
    ? undefined
    : new Store({
        plugins: [plugin(Object.values(modules))],
      })
