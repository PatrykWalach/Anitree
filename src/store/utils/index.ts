export * from './useCommands'
export * from './useTitle'
export * from './useEdit'
