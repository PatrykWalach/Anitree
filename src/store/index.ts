import Vue from 'vue'
import Vuex from 'vuex'

// import { getModule } from 'vuex-module-decorators'
Vue.use(Vuex)

import { ModuleMedia } from './modules/media'
import { ModuleTheme } from './modules/theme'
import { ModuleTitle } from './modules/title'
import { ModuleAuth } from './modules/auth'

export interface RootState {
  media: ModuleMedia
  theme: ModuleTheme
  title: ModuleTitle
  auth: ModuleAuth
}

export default new Vuex.Store<RootState>({})
