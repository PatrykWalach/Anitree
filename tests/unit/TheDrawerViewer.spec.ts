import TheDrawerViewer from '@/components/TheDrawerViewer.vue'
import { matchSnapshot } from './utils'
import { mockedViewer } from './mocks/viewer'
import { useMockedStore } from './mocks/store'

describe('TheDrawerViewer', () => {
  matchSnapshot(TheDrawerViewer, {
    propsData: {
      viewer: mockedViewer,
    },
    provide: { ...useMockedStore() },
  })
})
