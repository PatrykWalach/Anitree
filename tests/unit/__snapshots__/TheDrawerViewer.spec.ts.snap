// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDrawerViewer matches snapshot 1`] = `<!---->`;

exports[`TheDrawerViewer matches snapshot 2`] = `<!---->`;
